src/main/java/org/jenkins/plugins/lockableresources/LockableResourcesManager.java

```diff
@@ -119,6 +119,26 @@ public synchronized void setDeclaredResources(List<LockableResource> declaredRes
     this.resources = mergedResources;
   }
 
+  public void deleteResource(String name) {
+    LockableResource resource = fromName(name);
+    if (resource == null) {
+      return;
+    }
+
+    if (resource.isLocked()) {
+      // Removed locks became ephemeral.
+      resource.setDescription("");
+      resource.setLabels("");
+      resource.setNote("");
+      resource.setEphemeral(true);
```

Comment:
the user shall be inform about that
+ in the job log
+  in the UI like setting some other description. ( in this case I will not change note, labels) Just set ephemeral = true.

One hit more. What happens, when other job is waiting for this resource? This one will be on release deleted. The next one will create it again, or in worst case crashed the job.